core/src/main/java/io/parsingdata/metal/data/ParseState.java

```diff
@@ -42,61 +42,63 @@ public class ParseState extends ImmutableObject {
     public final Source source;
     public final ImmutableList<ImmutablePair<Token, BigInteger>> iterations;
     public final ImmutableList<ParseReference> references;
+    public final int scopeDepth;
 
-    public ParseState(final ParseGraph order, final ParseValueCache cache, final Source source, final BigInteger offset, final ImmutableList<ImmutablePair<Token, BigInteger>> iterations, final ImmutableList<ParseReference> references) {
+    public ParseState(final ParseGraph order, final ParseValueCache cache, final Source source, final BigInteger offset, final ImmutableList<ImmutablePair<Token, BigInteger>> iterations, final ImmutableList<ParseReference> references, final int scopeDepth) {
```

Comment:
A constructor with seven arguments, oof :sweat_smile:

I don't think we should address this now, but we should discuss this when/if we refactor the `ParseGraph` implementation.

Reply:
I just created #415 , where we are going to move the scopeDepth to the ParseGraph instead. This way, the `withOrder` method can be used correctly and the ParseState will not hold an incorrect scopeDepth. I developed it in our fork for now, but will create a separate PR here.